reference/gall.md

```diff
@@ -0,0 +1,765 @@
+# Gall
+
+## Reference Agent
+
+This app stores a number for each user. You can increment your number, increment
+someone else's number, start tracking someone else's number by poking. It
+exposes a scry namespace to inspect everyone's number.
+
+### `app/example-gall.hoon`
+
+```hoon
+/-  *example-gall
+/+  default-agent, verb
+|%
++$  card  card:agent:gall
++$  note
+  $%
+    [%arvo =note-arvo]
+    [%agent [=ship name=term] =task:agent:gall]
+  ==
++$  state  [local=@ud remote=(map @p @ud)]
+--
+=|  state
+=*  sta  -
+^-  agent:gall
+|_  =bowl:gall
++*  this  .
+    def   ~(. (default-agent this %|) bowl)
+::
+++  on-init
+  ^-  (quip card _this)
+  `this(local.sta 1)
+::
+++  on-save
+  !>(sta)
+::
+++  on-load
+  |=  old=vase
+  ^-  (quip card _this)
+  `this(sta !<(state old))
+::
+++  on-poke
+  |=  [=mark =vase]
+  ^-  (quip card _this)
+  ?>  =(mark %example-gall-action)
+  =/  act=example-gall-action  !<(example-gall-action vase)
+  ?-  act
+      ::
+    [%increment ~]
+  :-  [%give %fact `/local %atom !>(+(local.sta))]~
+  this(local.sta +(local.sta))
+      ::
+    [%increment-remote who=@p]
+  :_  this
+  :~  :*
+    %pass
+    /inc/(scot %p who.act)
+    %agent
+    [who.act %example-gall]
+    %poke  %example-gall-action  !>([%increment ~])
+  ==  ==
+      ::
+    [%view who=@p]
+  :_  this
+  [%pass /view/(scot %p who.act) %agent [who.act %example-gall] %watch /local]~
+      ::
+    [%stop-view who=@p]
+  :_  this(remote.sta (~(del by remote.sta) who.act))
+  [%pass /view/(scot %p who.act) %agent [who.act %example-gall] %leave ~]~
+      ::
+  ==
+::
+++  on-leave
+  |=  =path
+  ^-  (quip card _this)
+  ~&  "Unsubscribed by: {<src.bowl>} on: {<path>}"
+  `this
+::
+++  on-watch
+  |=  =path
+  ^-  (quip card _this)
+  :_  this
+  ~&  path
+  ?+  path  ~
+       ::
+     [%local ~]
+  [%give %fact ~ %atom !>(local.sta)]~
+  ==
+::
+++  on-peek
+  |=  =path
+  ^-  (unit (unit cage))
+  ?+  path  [~ ~]
+      ::
+    [%x %local ~]
+  ``[%atom !>(local.sta)]
+      ::
+    [%x %remote who=@ta ~]
+  ~&  remote.sta
+  =/  res  (~(got by remote.sta) (slav %p -.+.+.path))
```

Comment:
Path part access generally happens as `i.t.t.path`. It isn't much better, but at least it's faces. You might also want to `=*  location  i.t.t.path` on the line before, for clarity.